src/app/mypage/edit/page.tsx

```diff
@@ -0,0 +1,46 @@
+'use client'
+import UserProfile from '@/components/UserContainer/UserProfile'
+import UserEditButton from '@/components/UserContainer/UserEditButton'
+import Button from '@/components/common/Button'
+import { useState } from 'react'
+
+export default function MyPageEdit() {
+  const [isEditMode, setIsEditMode] = useState(false)
+  const userInfo = {
+    nickName: 'gaori',
+    profile: '/image/default_profile.svg',
+  }
+  const onClickPencil = () => {
+    setIsEditMode(!isEditMode)
+    alert('pencil')
+  }
+  return (
+    <main className="w-full h-screen flex items-center flex-col pt-[90px]">
+      <div className="w-[100px] h-[100px] relative">
+        <UserProfile isEditable={true} size={100} />
+      </div>
+      <form
+        action=""
+        className="w-[230px] p-[10px] mt-[70px] mb-[20px] flex justify-between flex-wrap border-b-[2.5px] border-b-brown font-sindinaru-b text-brown"
+      >
+        <label htmlFor="nickname" className="w-full block mb-[5px]">
+          이름
+        </label>
+        <input
+          type="text"
+          name="nickname"
+          id="nickname"
+          placeholder="이름을 입력해주세요"
+          maxLength={10}
+          className="w-[calc(100% - 25px)] placeholder:text-gray bg-inherit"
+        />
+        {!isEditMode && <UserEditButton type="pencil" onClick={() => onClickPencil()} />}
```

Comment:
```suggestion
        {isEditMode || <UserEditButton type="pencil" onClick={() => onClickPencil()} />}
```
이렇게 쓰는 게 더 직관적일 것 같아용
수정해달라는 건 아니고... 참고하세요~

Reply:
혹시 언니 천재..?